File: 0230-kth-smallest-element-in-a-bst/0230-kth-smallest-element-in-a-bst.java
/**
 * Definition for a binary tree node.
 * public class TreeNode {
 *     int val;
 *     TreeNode left;
 *     TreeNode right;
 *     TreeNode() {}
 *     TreeNode(int val) { this.val = val; }
 *     TreeNode(int val, TreeNode left, TreeNode right) {
 *         this.val = val;
 *         this.left = left;
 *         this.right = right;
 *     }
 * }
 */
class Solution {
    List<Integer> inorder = new ArrayList<>();
	
    public int kthSmallest(TreeNode root, int k) {
        printInorder(root, k);
        return inorder.get(k-1);
    }
	
    public void printInorder(TreeNode root, int k) {
        if(root != null && inorder.size() < k) {
            printInorder(root.left, k);
            inorder.add(root.val);
            printInorder(root.right, k);
        }
        
    }
}

File: 0230-kth-smallest-element-in-a-bst/0230-kth-smallest-element-in-a-bst.py
# Definition for a binary tree node.
# class TreeNode:
#     def __init__(self, val=0, left=None, right=None):
#         self.val = val
#         self.left = left
#         self.right = right
class Solution:
    def kthSmallest(self, root: Optional[TreeNode], k: int) -> int:
        if not root:
            return -1

        count =0
        result = None

        def helper(root):
            nonlocal count,result
            if not root:
                return None
            helper(root.left)
            count= count+1
            if count == k:
                result = root.val
                return 
            helper(root.right)
        helper(root)
        return result
            
            

        